plugins/catalog-backend/api-report.md

```diff
@@ -167,6 +167,12 @@ export class CatalogBuilder {
   useLegacySingleProcessorValidation(): this;
 }
 
+// @public (undocumented)
+export interface CatalogCollatorEntityTransformer {
+  // (undocumented)
+  transform(entity: Entity, locationTemplate: string): CatalogEntityDocument;
+}
```

Comment:
how would you feel about making this

```ts
export type CatalogCollatorEntityTransformer = 
  (entity: Entity) => CatalogEntityDocument;
```

instead? We use functions in the other similar places, and there's little value in forcing people to make a wrapping class around the callback.

Possibly even making it return `Omit<CatalogEntityDocument, 'location'>` as per the other comment.

Reply:
Also, we mentioned publicly exporting a `defaultCatalogCollatorEntityTransformer` function here, so that people can easily call that and then just modify one single field for example, instead of reinventing the entire output

Reply:
I think there is no difference between both options, either you can call parent method or `default` function, so no difference, just class interface will make it possible to extend interface if we will need in future

Reply:
Yeah, I'd be more comfortable with a function vs. a class as well.  My worry with the class-based approach is that it implies that it can or should do much more than just map values to fields; it should never go as far as to be constructed with dependencies that can then be used to do more complicated things (like network IO, etc).  ...Decorators were introduced to handle use-cases like that.

I might also recommend an alternate approach, here...  Where instead of exporting `defaultCatalogCollatorEntityTransformer` and having users call it to get the default behavior (on top of which, they could make their own tweaks)...  Instead, just continue to perform the existing transformations in the collator, _then_ call any user-provided `EntityTransformer`, merging in any/all values, and _then_ applying things that should not be overridable via this mechanism (e.g. the `location` and `authorization`, as @freben mentioned).

That would allow us to...

- Maintain existing behavior
- Make it very easy to _replace_ the existing mapping
- Make it very easy to _extend_ the existing mapping (without having to know to call something else)
- Enforce constraints

Reply:
refactored to function, I think it would be better if custom function which will be passed will override default one so we will have easier control over what we go inside document, otherwise in case we will want no to add sth from default one we will need explicitly specify it as undefined which is a place for a lot of bugs (forgot to add sth etc)